shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,18 +226,187 @@ async fn fetch_current_nodes(
+                    "keyspace query returned unexpected cassandra operation: {:?}",
+                    operation
+                )),
+            }
+        } else {
+            Err(anyhow!("Failed to parse keyspace query response"))
+        }
+    }
+
+    fn build_keyspace(
+        row: &mut Vec<MessageValue>,
+        data_center: &str,
+    ) -> Result<(String, KeyspaceMetadata)> {
+        let metadata = if let Some(MessageValue::Varchar(string)) = row.pop() {
+            let replication: JsonValue = serde_json::from_str(&string).map_err(|error| {
+                anyhow!(format!(
+                    "Error parsing replication. Error: {} Replication: {}",
+                    error, string
+                ))
+            })?;
+
+            let replication_strategy = match replication {
+                JsonValue::Object(properties) => build_replication_strategy(properties)?,
+                _ => {
+                    return Err(anyhow!(format!(
+                        "Error parsing replication strategy: {}",
+                        replication
+                    )))
+                }
+            };
+
+            let replication_factor = match replication_strategy {
+                ReplicationStrategy::SimpleStrategy { replication_factor } => replication_factor,
+                ReplicationStrategy::NetworkTopologyStrategy {
+                    datacenter_replication_factor,
+                } => *datacenter_replication_factor.get(data_center).unwrap_or(&0),
+                _ => 0,
+            };
+
+            KeyspaceMetadata { replication_factor }
+        } else {
+            return Err(anyhow!(
+                "system_schema.keyspaces.replication is not a varchar"
+            ));
+        };
+
+        let name = if let Some(MessageValue::Varchar(name)) = row.pop() {
+            name
+        } else {
+            return Err(anyhow!("system_schema_keyspaces.name"));
+        };
+
+        Ok((name, metadata))
+    }
+
+    fn build_replication_strategy(
+        mut properties: Map<String, JsonValue>,
+    ) -> Result<ReplicationStrategy> {
+        match properties.remove("class") {
+            Some(JsonValue::String(class)) => Ok(match class.as_str() {
+                "org.apache.cassandra.locator.SimpleStrategy" | "SimpleStrategy" => {
+                    ReplicationStrategy::SimpleStrategy {
+                        replication_factor: extract_replication_factor(
+                            properties.get("replication_factor"),
+                        )?,
+                    }
+                }
+                "org.apache.cassandra.locator.NetworkTopologyStrategy"
+                | "NetworkTopologyStrategy" => ReplicationStrategy::NetworkTopologyStrategy {
+                    datacenter_replication_factor: extract_datacenter_replication_factor(
+                        properties,
+                    )?,
+                },
+                _ => ReplicationStrategy::Other,
```

Comment:
Is this case just taken when we havent implemented support for a replication strategy yet?
If thats the case we should emit an `error!` and rename `ReplicationStrategy::Other` to `ReplicationStategy::Unknown`

Reply:
Closing and reopening on a more recent line